src/fu-engine.c

```diff
@@ -5723,11 +5724,14 @@ fu_engine_update_history_device (FuEngine *self, FuDevice *dev_history, GError *
 	if (fu_device_get_update_state (dev) != FWUPD_UPDATE_STATE_FAILED &&
 	    fu_device_get_update_state (dev) != FWUPD_UPDATE_STATE_FAILED_TRANSIENT) {
 		g_debug ("falling back to generic failure");
+		fu_device_set_update_state (dev_history, FWUPD_UPDATE_STATE_FAILED);
```

Comment:
do you know what state is actually coming into this method?  If it's not failed and not failed-transient, doesn't that mean it's a plugin bug in not setting it?

Reply:
If we're calling into GetResults vfunc (like the UEFI plugin) then the plugin doesn't know if a firmware was deployed, scheduled or whatever and just returns the ESRT value. In the case where we've just launched fwupd and the pending update firmware does not match the running firmware version then returning success is "correct" but not true.

My worry here is that we schedule an update that doesn't get run (e.g. because the user has set up custom secure boot keys, or deliberately aborts the flashing firmware and we'd record a hard failure, not `FAILED_TRANSIENT`. Of course In some cases we do want a hard `FAILED` status where the capsule was run, but didn't update the `Status` value.

Reply:
I guess relying upon strictly the ESRT status is not enough because users can put other items into the flow that break things like you say.